lib/yardstick/rule_description/classifier.rb

```diff
@@ -0,0 +1,105 @@
+module Yardstick
+  # Rule description composed of tokens which can be formatted
+  class RuleDescription
+    # Token classification
+    module Classifier
+      # List of classifiers
+      class List
+        extend Forwardable
+        include Concord.new(:classifiers), Enumerable
+
+        # Classify a token by returning the first match
+        #
+        # @param token [String] markup text
+        #
+        # @return [Token]
+        #
+        # @api private
+        def classify(token)
+          classifiers.each do |classifier|
+            next unless (classified = classifier.try(token))
+
+            return classified
+          end
+        end
+
+        # Union pattern for splitting text based on patterns
+        #
+        # @return [Regexp]
+        #
+        # @api private
+        def pattern
+          Regexp.union(pattern_classifiers.map(&:pattern))
+        end
+
+        private
```

Comment:
This is a small nitpick, but I typically outdent my `private` and `protected` keywords in classes and modules.